pkg/cli/apk.go

```diff
@@ -221,14 +189,124 @@ func cmdCp() *cobra.Command { //nolint:gocyclo
 	return cmd
 }
 
-func addAuth(u *url.URL) {
+func cmdApkLs() *cobra.Command {
+	var full bool
+	var latest bool
+	var j bool
+	var packageFilter string
+
+	cmd := &cobra.Command{
+		Use:     "ls",
+		Example: `wolfictl apk ls https://packages.wolfi.dev/os/x86_64/APKINDEX.tar.gz`,
+		Args:    cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+			u := args[0]
+
+			dir := strings.TrimSuffix(u, "/APKINDEX.tar.gz")
+
+			index, _, err := fetchAPKIndex(ctx, u)
+			if err != nil {
+				return err
+			}
+
+			w := cmd.OutOrStdout()
+			enc := json.NewEncoder(w)
+
+			packages := index.Packages
+
+			// TODO: origin filter as well?
+			if packageFilter != "" {
+				packages = slices.DeleteFunc(packages, func(pkg *apk.Package) bool {
+					return pkg.Name != packageFilter
+				})
+			}
+
+			if latest {
+				packages = onlyLatest(packages)
+			}
+
+			for _, pkg := range packages {
+				p := fmt.Sprintf("%s-%s.apk", pkg.Name, pkg.Version)
+				u := fmt.Sprintf("%s/%s", dir, p)
+				switch {
+				case j:
+					if err := enc.Encode(pkg); err != nil {
+						return fmt.Errorf("encoding %s: %w", pkg.Name, err)
+					}
+				case full:
+					fmt.Fprintf(w, "%s\n", u)
+				default:
+					fmt.Fprintf(w, "%s\n", p)
+				}
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVarP(&packageFilter, "package", "P", "", "print only packages with the given name")
+	cmd.Flags().BoolVar(&latest, "latest", false, "print only the latest version of each package")
+	cmd.Flags().BoolVar(&full, "full", false, "print the full url or path")
+	cmd.Flags().BoolVar(&j, "json", false, "print each package as json")
+
+	return cmd
+}
+
+func fetchAPKIndex(ctx context.Context, indexURL string) (*apk.APKIndex, string, error) {
+	var arch string
+	repoURL := strings.TrimSuffix(indexURL, "/APKINDEX.tar.gz")
+	var in io.ReadCloser
+	switch {
+	case indexURL == "-":
+		in = os.Stdin
+		arch = "x86_64" // TODO: This is hardcoded.
+	case strings.HasPrefix(indexURL, "file://"):
```

Comment:
I know these were here before, just curious if we use these, or just have them here because the seem good?

Reply:
Mostly just because they seem good. I'm not married to it if we decide to drop it now/later.